src/option.rs

```diff
@@ -1,24 +1,26 @@
-//! Option Support
-//!
-//! # Design
-//!
-//! Miniconf supports optional values in two forms. The first for is the [`Option`] type. If the
-//! `Option` is `None`, the part of the namespace does not exist at run-time.
-//! It will not be iterated over and cannot be `get()` or `set()` using the Miniconf API.
-//!
-//! This is intended as a mechanism to provide run-time construction of the namespace. In some
-//! cases, run-time detection may indicate that some component is not present. In this case,
-//! namespaces will not be exposed for it.
-//!
-//!
-//! # Standard Options
-//!
-//! Miniconf also allows for the normal usage of Rust `Option` types. In this case, the `Option`
-//! can be used to atomically access the nullable content within.
-
 use super::{Error, IterError, Metadata, Miniconf, Peekable};
-
-/// An `Option` that exposes its value through their [`Miniconf`](trait.Miniconf.html) implementation.
+use core::ops::{Deref, DerefMut};
+
+/// An `Option` that exposes its value through their [`Miniconf`] implementation.
+///
+/// # Design
+///
+/// Miniconf supports optional values in two forms. The first for is the [`miniconf::Option`](Option)
+/// type. If the `Option` is `None`, the part of the namespace does not exist at run-time.
+/// It will not be iterated over and cannot be `get()` or `set()` using the [`Miniconf`] API.
```

Comment:
Isn't this doc no longer true? i.e. you will always iterate over `None` topic trees regardless, but they generate a unique error during `set()` and `get()`